massa-models/src/slot.rs

```diff
@@ -153,6 +153,68 @@ impl Slot {
     }
 }
 
+use crate::serialization::DeserializeVarIntV2;
+use nom::{bytes::complete::take, error::context, sequence::tuple, IResult};
+
+fn take_thread_number(s: &[u8]) -> IResult<&[u8], u8> {
+    match take(1usize)(s) {
+        Ok((rest, v)) => {
+            let thread_number = v[0];
+            if thread_number < THREAD_COUNT {
+                Ok((rest, thread_number))
+            } else {
+                Err(nom::Err::Failure(nom::error::Error::new(
+                    s,
+                    nom::error::ErrorKind::Fail,
+                )))
+            }
+        }
+        Err(e) => Err(e),
+    }
+}
+
+// HERE
+impl DeserializeCompactV2 for Slot {
+    fn from_bytes_compact_v2<'a>(buffer: &'a [u8]) -> IResult<&[u8], Self> {
+        tuple((
+            context("period", |buffer: &'a [u8]| {
```

Comment:
Is that a question or a request?

If what you are saying is that `u64::from_varint_bytes_v2` should return a `FnOnce`, yes I agree it would be better.

Reply:
It's a request. I think we should be able to easily write the deserialization method and all fields should have the same format to deserialize and the same as nom.